src/Package/Impl/DataInspect/VisualGrid/MatrixViewAutomationPeerScrollProvider.cs

```diff
@@ -0,0 +1,107 @@
+        }
+
+        public double HorizontalViewSize {
+            get => _horizontalViewSize;
+            private set => SetProperty(ref _horizontalViewSize, value, HorizontalViewSizeProperty);
+        }
+
+        public double VerticalViewSize {
+            get => _verticalViewSize;
+            private set => SetProperty(ref _verticalViewSize, value, VerticalViewSizeProperty);
+        }
+
+        public bool HorizontallyScrollable {
+            get => _horizontallyScrollable;
+            private set => SetProperty(ref _horizontallyScrollable, value, HorizontallyScrollableProperty);
+        }
+
+        public bool VerticallyScrollable {
+            get => _verticallyScrollable;
+            private set => SetProperty(ref _verticallyScrollable, value, VerticallyScrollableProperty);
+        }
+
+        public MatrixViewAutomationPeerScrollProvider(MatrixView owner) : base(owner) {
+            _owner = owner;
+        }
+
+        public void UpdateValues() {
+            var points = _owner.Points;
+            HorizontallyScrollable = points.HorizontalExtent.GreaterThan(points.ViewportWidth);
+            VerticallyScrollable = points.VerticalExtent.GreaterThan(points.ViewportHeight);
+            HorizontalViewSize = HorizontallyScrollable ? 100 * points.ViewportWidth / points.HorizontalExtent : 100;
+            VerticalViewSize = VerticallyScrollable ? 100 * points.ViewportHeight / points.VerticalExtent : 100;
+            HorizontalScrollPercent = HorizontallyScrollable
+                ? 100 * points.HorizontalOffset / (points.HorizontalExtent - points.ViewportWidth)
+                : NoScroll;
+            VerticalScrollPercent = VerticallyScrollable
+                ? 100 * points.VerticalOffset / (points.VerticalExtent - points.ViewportHeight)
+                : NoScroll;
+        }
+
+        public void Scroll(ScrollAmount horizontalAmount, ScrollAmount verticalAmount) {
+            switch (horizontalAmount) {
+                case ScrollAmount.LargeDecrement:
+                    Scroller.EnqueueCommand(GridUpdateType.PageLeft, 1);
+                    break;
+                case ScrollAmount.SmallDecrement:
+                    _owner.Scroller.EnqueueCommand(GridUpdateType.LineLeft, 1);
+                    break;
+                case ScrollAmount.SmallIncrement:
+                    _owner.Scroller.EnqueueCommand(GridUpdateType.LineRight, 1);
+                    break;
+                case ScrollAmount.LargeIncrement:
+                    _owner.Scroller.EnqueueCommand(GridUpdateType.PageRight, 1);
+                    break;
+            }
+
+            switch (verticalAmount) {
+                case ScrollAmount.LargeDecrement:
+                    _owner.Scroller.EnqueueCommand(GridUpdateType.PageUp, 1);
+                    break;
+                case ScrollAmount.SmallDecrement:
+                    _owner.Scroller.EnqueueCommand(GridUpdateType.LineUp, 1);
```

Comment:
It is `null` if user changes variable to something that we can't display, for example:
1. `x <- data.frame(replicate(1000, sample(0:1000, 1000, rep = TRUE)))`
2. Open variable view
3. `x <- function() {}`